### PR TITLE
Cache the file seen status

### PR DESCRIPTION
**Pull request by josephwright**
**READ ME FIRST**: Please understand that in most cases we will not be able to merge a pull request because there are a lot of internal activities needed when updating the LaTeX2e sources. If you have a code suggestion please discuss it with the team first.

*Pull requests in this repository are intended for LaTeX Team members only.*

# Internal housekeeping

## Status of pull request

- Feedback wanted 
- Under development
- [X] Ready to merge

## Checklist of required changes before merge will be approved
- [ ] Test file(s) added
- [X] Version and date string updated in changed source files
- [X] Relevant `\changes` entries in source included
- [X] Relevant `changes.txt` updated
- [X] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

### CODE REVIEW

File: base/changes.txt [hunk 1]
```diff
@@ -6,6 +6,11 @@ completeness or accuracy and it contains some references to files that
 are not part of the distribution.
 ================================================================================
 
+2023-05-10  Joseph Wright  <Joseph.Wright@latex-project.org>
+
+	* lffiles.dtx:
+	Cache the status of files seen
```
Comment (FrankMittelbach) [inline base/changes.txt:12]: should list the related issue/pr

**Review by FrankMittelbach — APPROVED** [review]

File: base/changes.txt (hunk 1 above)
Comment (josephwright) [reply base/changes.txt:12]: I can't see one - can you find it?

**Comment by josephwright** [comment]
@davidcarlisle Doe the code here look OK?

**Review by davidcarlisle — APPROVED** [review]

**Comment by davidcarlisle** [comment]
The failing tests all fail for the same issue on whether files specified without extension are associated with `foo` or `foo.tex` eg github-0464

`\AddToHook{file/after}{\typeout{After "\CurrentFile"}}`

```
*** ../build/test-config-lthooks/github-0464.tlg	Sun May 14 11:24:51 2023
--- ../build/test-config-lthooks/github-0464.pdftex.log	Sun May 14 11:24:52 2023
***************
*** 1,4 ****
  This is a generated file for the l3build validation system.
  Don't change this file in any respect.
  (testA.tex)
! After "testA.tex"
--- 1,4 ----
  This is a generated file for the l3build validation system.
  Don't change this file in any respect.
  (testA.tex)
! After "testA"
```

**Pull request closed without merging**